apps/myaccount/src/hooks/use-organizations.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import useAppSettings from "./use-app-settings";
+import { MultiTenantConstants } from "../constants/multi-tenant-constants";
+
+/**
+ * Interface for the return type of the UseOrganizations hook.
+ */
+export interface UseOrganizationsInterface {
+    /**
+     * Set the organization id in the local storage.
+     * @param orgId - Organization id.
+     */
+    setOrgIdInLocalStorage: (orgId: string) => void;
+    /**
+     * Set the user's organization in the local storage.
+     * @param userOrg - User's organization.
+     */
+    /**
+     * Sets the user organization in local storage.
+     * @param userOrg - The user organization to set.
+     */
+    setUserOrgInLocalStorage: (userOrg: string) => void;
+    /**
+     * Retrieves the organization ID from local storage.
+     */
+    getOrgIdInLocalStorage: () => string;
+    /**
+     * Retrieves the user organization from local storage.
+     */
+    getUserOrgInLocalStorage: () => string;
+    /**
+     * Removes the organization ID from local storage.
+     */
+    removeOrgIdInLocalStorage: () => void;
+    /**
+     * Removes the user organization from local storage.
+     */
+    removeUserOrgInLocalStorage: () => void;
+    /**
+     * Transforms the tenant domain to the correct format.
+     *
+     * @param tenantDomain - Tenant domain.
+     * @returns Transformed tenant domain.
+     */
+    transformTenantDomain: (tenantDomain: string) => string;
+}
+
+/**
+ * Hook that provides access to the Organizations context.
+ *
+ * @returns An object containing the current Organizations context.
+ */
+const useOrganizations = (): UseOrganizationsInterface => {
+    const { getLocalStorageSetting, setLocalStorageSetting, removeLocalStorageSetting } = useAppSettings();
+
+    /**
+     * Transforms the tenant domain to the correct format.
+     *
+     * @param tenantDomain - Tenant domain.
+     * @returns Transformed tenant domain.
+     */
+    const transformTenantDomain = (tenantDomain: string): string => {
+        // With the latest Authz framework, `carbon.super` is resolved as `Super`.
```

Comment:
Will we need to add legacy runtime check here, given that this change only works as expected for the new authz runtime, as per the comment?

Reply:
I don't think we need to add the legacy check here since it's not expected to call this method in the legacy mode.